### DISCUSSION TITLE
Tokened links that could be done - The Thread

### DISCUSSION THREAD

**Discussion by BellezaEmporium** [iptv-org/community#101] [Show and tell]
This will reference, for me, certain links that can be fixable easily (that I can't do for now)

EDIT : i'm slowly merging all the work i've done from "iptv-all" to streamlink plugins.

RO : 

- [x] ZU TV = https://radiozu.ro/live?_pjax=%23pjax-container (https://iptv-all.lanesh4d0w.codes/romania/zutv)

(released workaround in last push)

From Sphinxroot : 

SN : 

- [x] 7TV = http://7tv.sn/livetv/sevenlive (https://iptv-all.lanesh4d0w.codes/senegal/7tv)
- [x] DTV = http://dtv.sn/home/dtvlive (https://iptv-all.lanesh4d0w.codes/senegal/dtv)
- [x] Label TV = http://www.labeltelevision.com/direct (https://iptv-all.lanesh4d0w.codes/senegal/label-tv)
- [x] SenTv = https://sentv.sn/direct (https://iptv-all.lanesh4d0w.codes/senegal/sentv)

SR : 

- [ ] atv = http://www.atv.sr/live-broadcast/

AE : 

SSC 1 (MBC Shahid VIP) : https://shls-ssc-1-prod-dub.shahid.net/out/v1/0da39f594cc34c4a9b03f0d696a86276/index.mpd + https://drm.cloud.insysvt.com/acquire-license/widevine?BrandGuid=2be49af0-6fbd-4511-8e11-3d6523185bb4&UserToken=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJleHAiOjEuNjMzMDgzMDc2RTksImRybVRva2VuSW5mbyI6eyJleHAiOiIyMDIxLTEwLTAxVDEwOjExOjE2LjAwMDE4OSIsImtpZCI6WyIqIl0sInAiOnsicGVycyI6ZmFsc2UsImVkIjoiMjAyMS0xMC0wMVQxMDoxMToxNi4wMDAxODkiLCJleGMiOnsiRmFpcnBsYXlSZW50YWxEdXJhdGlvblNlY29uZHMiOjg2MDAwLCJGYWlycGxheUxlYXNlRHVyYXRpb25TZWNvbmRzIjo4NjAwMH19fX0.DuX2g6AAH2FLq-btrqeMjnjiqnmShS6Nxh_6kH_zEcU

GT ;

https://www.chapintv.com/envivo-canal-3/ <- NedMedia token => rsk from script most certainly.

**Comment by Sphinxroot** [comment]
#4689  hope you can do something with those :)

**Comment by Sphinxroot** [comment]
will need to create sr.m3u
# http://www.atv.sr/live-broadcast/
#EXTINF:-1 tvg-id="atv.sr" tvg-country="SR" tvg-language="Dutch;English" tvg-logo="" group-title="",atv
https://edge1.tikilive.com/livehttporigin_tikilive/34967/amlst:mainstream/playlist.m3u8 +Token

if something can be done thanks :)

**Reply by BellezaEmporium** [reply]
https://edge1.tikilive.com/livehttporigin_tikilive/34967/amlst:mainstream/playlist.m3u8?p=local&op_id=19&userId=0&channelId=34967&stime=1633860842&etime=1633947242&token=0d92e3a5c6b23daf5925e

Can't seem to get anything, seems the link is given via socket. Code gives no results.

**Comment by Sphinxroot** [comment]
https://live.rts.sn/home + 3 News Channel to come  RTS3 RTS4 RSI
https://live.rts.sn/livetv/rts1 = https://iptv-all.lanesh4d0w.codes/senegal/rts1

#EXTINF:-1,RTS 1
https://live1.acangroup.org:1929/larts/rts1.stream/playlist.m3u8 + ?wmsAuthSign= BASE64 Token=  "server_time= (DATE+TIME) +  &hash_value= md5IP &validminutes=10"

https://live.rts.sn/livetv/rts2 = https://iptv-all.lanesh4d0w.codes/senegal/rts2
#EXTINF:-1,RTS 2
https://live1.acangroup.org:1929/larts/rts2.stream/playlist.m3u8 + ?wmsAuthSign= BASE64 Token=  "server_time= (DATE+TIME) +  &hash_value= md5IP &validminutes=10"

**Comment by BellezaEmporium** [comment]
@Sphinxroot RTS is done, got the changes locally, it seems the group behind their streams use the same session code for everything, so perfect for us haha.

EDIT : Changes pushed and managed to pass through the filthy websites using the SenTV API

**Reply by BellezaEmporium** [reply]
Prepared placeholders for RTS3 and RTS4. RSI wasn't added because it's a radio.

**Comment by SSyar** [comment]
do something about this https://www.mjunoon.tv/watch-live-tv-channels

**Reply by BellezaEmporium** [reply]
> 
> 
> do something about this https://www.mjunoon.tv/watch-live-tv-channels

It seems like it's transiting through a POST on https://beta.mjunoon.tv:9191/v2/streaming-url and it gives a loooong URL, encoded idk how.

**Reply by SSyar** [reply]
> 
> 
> > do something about this https://www.mjunoon.tv/watch-live-tv-channels
> 
> It seems like it's transiting through a POST on https://beta.mjunoon.tv:9191/v2/streaming-url and it gives a loooong URL, encoded idk how.

can it be hotlinked .. .!!

**Reply by jkemp814** [reply]
Chrome prompts to install extension. (mjunoon.tv) 

Once the 'chrome app' is install I selected CNN and got error:
This stream may not be available in your country.
(Error Code : 4031)

Quite a few of the other channels play.

https://www.mjunoon.tv/news-one-live works in Firefox once the URL is copied from 'chrome app'

**Reply by SSyar** [reply]
its working on the browser but can the url be used in IPTV application !!

**Reply by jkemp814** [reply]
> its working on the browser but can the url be used in IPTV application !!

Are you yelling at me for trying to add some info? (double exclamation marks)

**Reply by BellezaEmporium** [reply]
I don't have the extension pop up on my side (using Firefox), but that won't seem to be an easy task. Might be luckier if I directly try to catch the link from the player (pretty sure it's gonna be IP-locked)

**Reply by jkemp814** [reply]
Yes, I did the same thing. Kept getting 403 forbidden.

**Reply by BellezaEmporium** [reply]
It seems... that the link is encrypted and decrypted after by the player by some function that I don't know.

**Reply by SSyar** [reply]
> 
> 
> > its working on the browser but can the url be used in IPTV application !!
> 
> Are you yelling at me for trying to add some info? (double exclamation marks)

!! is curiosity

 

> 
> 
> It seems... that the link is encrypted and decrypted after by the player by some function that I don't know.

and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=

**Reply by jkemp814** [reply]
> !! is curiosity

Then I suggest you use ?, instead of !

**Reply by BellezaEmporium** [reply]
IP-locked it seems. Sadly that won't work.

**Reply by Sphinxroot** [reply]
> and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=

?wmsAuthSign= "base64 Access Token" (server_time="DATE+TIME"&hash_value="md5IP"&validminutes=1)

**Reply by BellezaEmporium** [reply]
> 
> 
> > and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=
> 
> ?wmsAuthSign= "base64 Access Token" (server_time="DATE+TIME"&hash_value="md5IP"&validminutes=1)

Hmm... that might be doable with the actual person's IP if I can get it from my server... let me see.

**Reply by BellezaEmporium** [reply]
I have a weird something.
c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE= = server_time=10/7/2021 5:32:48 AM&hash_value=trrpyPdLhslja9Qu5WJ1pA==&validminutes=1

Hash Value is UTF-8 "Kck.bu"

**Reply by BellezaEmporium** [reply]
> 
> 
> > and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=
> 
> ?wmsAuthSign= "base64 Access Token" (server_time="DATE+TIME"&hash_value="md5IP"&validminutes=1)

Tried this, doesn't seem to do anything.

**Reply by BellezaEmporium** [reply]
Here's the (dirty !) code I've made to try to get the link work (without any positive result)

EDIT : Might have "cracked" the way it works.

```
app.get('/testings/mjunoon', function(req,res) {
    const forwarded = req.headers['x-forwarded-for']
    const ip = forwarded ? forwarded.split(/, /)[0] : req.socket.remoteAddress
    var moment = require('moment-timezone');
    var good_date = moment().tz("Atlantic/Azores").format("MM/DD/YYYY hh:mm:ss A");
    var hmm2 = Base64.atob(ip);
    var theF = Base64.encode(hmm2)
    const construct_stuff = "server_time=" + good_date + "&hash_value=" + theF + "&validminutes=1"
    console.log(construct_stuff)
    var hmm = Base64.encode(construct_stuff);
    res.redirect('https://cdn7.mjunoon.tv:8087/web/Madni-121/playlist.m3u8?wmsAuthSign=' + hmm)
    console.log(hmm)
})
```

**Reply by SSyar** [reply]
> 
> 
> Here's the (dirty !) code I've made to try to get the link work (without any positive result)
> 
> EDIT : Might have "cracked" the way it works.
> 
> ```
> app.get('/testings/mjunoon', function(req,res) {
>     const forwarded = req.headers['x-forwarded-for']
>     const ip = forwarded ? forwarded.split(/, /)[0] : req.socket.remoteAddress
>     var moment = require('moment-timezone');
>     var good_date = moment().tz("Atlantic/Azores").format("MM/DD/YYYY hh:mm:ss A");
>     var hmm2 = Base64.atob(ip);
>     var theF = Base64.encode(hmm2)
>     const construct_stuff = "server_time=" + good_date + "&hash_value=" + theF + "&validminutes=1"
>     console.log(construct_stuff)
>     var hmm = Base64.encode(construct_stuff);
>     res.redirect('https://cdn7.mjunoon.tv:8087/web/Madni-121/playlist.m3u8?wmsAuthSign=' + hmm)
>     console.log(hmm)
> })
> ```

woAh,, so how can we use it in the iptv app ?

**Reply by SSyar** [reply]
> 
> 
> > > and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=
> > 
> > 
> > ?wmsAuthSign= "base64 Access Token" (server_time="DATE+TIME"&hash_value="md5IP"&validminutes=1)
> 
> Tried this, doesn't seem to do anything.

must hv expired by the time you accessed it

**Reply by BellezaEmporium** [reply]
> 
> 
> > > > and this how the link is created https://cdn6.mjunoon.tv:8087/web/SaudiQuran-142/playlist.m3u8?wmsAuthSign=c2VydmVyX3RpbWU9MTAvNy8yMDIxIDU6MzI6NDggQU0maGFzaF92YWx1ZT10cnJweVBkTGhzbGphOVF1NVdKMXBBPT0mdmFsaWRtaW51dGVzPTE=
> > > 
> > > 
> > > ?wmsAuthSign= "base64 Access Token" (server_time="DATE+TIME"&hash_value="md5IP"&validminutes=1)
> > 
> > 
> > Tried this, doesn't seem to do anything.
> 
> must hv expired by the time you accessed it

That would be odd, since the code above replicates what the server does. So the token's always fresh.

**Reply by SSyar** [reply]
> > > Tried this, doesn't seem to do anything.
> > 
> > 
> > must hv expired by the time you accessed it
> 
> That would be odd, since the code above replicates what the server does. So the token's always fresh.

i meant to say that specific llink with the token., could ve expired

**Reply by BellezaEmporium** [reply]
> do something about this https://www.mjunoon.tv/watch-live-tv-channels

Reviving old things, streamlink has a plugin for that, but it will only work locally (wmsAuthSign IP-locked)

**Comment by Sphinxroot** [comment]
.ch

https://www.auftanken.tv/livestream/   https://webplayer.sbctv.ch/auftanken/

https://tvoosa-pez0301.sctv.ch/http_adaptive_streaming/CMMLCKAAODKOMHBG.m3u8

CMMLCKAAODKOMHBG Will Change if stoped a new one is needed

seem that the Server player give a new id when loaded
MediaKind VSPP Streamer (Version: 7.3.0.0 Commit_id: 28e37d32d7f38f728f6d9b664ca0d4e3e95c881d Commit_time: 1617625253)
https://webplayer.sbctv.ch/THEOplayer/THEOplayer.js
https://webplayer.sbctv.ch/THEOplayer/theoplayer.e.js
https://webplayer.sbctv.ch/THEOplayer/theoplayer.p.js

**Reply by BellezaEmporium** [reply]
https://iptv-all.lanesh4d0w.codes/switzerland/sbctv

**Comment by Sphinxroot** [comment]
https://www.htatv.com/

**Reply by BellezaEmporium** [reply]
https://iptv-all.lanesh4d0w.codes/algeria/hta_all_in_1?channel= *insert channel*

**Comment by Sphinxroot** [comment]
https://cnnportugal.iol.pt/direto

**Reply by BellezaEmporium** [reply]
```
app.get("/portugal/cnn-pt", function(req,res) {
    fetch("https://services.iol.pt/matrix?userId=")
        .then(response => response.text())
        .then(text => {
            res.redirect("https://video-auth7.iol.pt/live_cnn/live_cnn/playlist.m3u8?wmsAuthSign=" + text)
        })
});
```

**Comment by SSyar** [comment]
why m3u8 link from these sites doesn't work (not tokened).. is it the referral ?
https://live.arydigital.tv/
https://live.arynews.tv/pk/
https://cricplay2.xyz/update/sscricket.php
https://cricplay2.xyz/update/astrocricket.php
http://cricplay2.xyz/update/fox501.php

**Reply by BellezaEmporium** [reply]
AryDigital would be https://6zklx4wryw9b-hls-live.5centscdn.com/arydigital/f7b44cfafd5c52223d5498196c8a2e7b.sdp/playlist.m3u8 / referrer locked to https://live.arydigital.tv/

![vlcsnap-2021-11-24-09h19m12s478](https://user-images.githubusercontent.com/30985701/143192638-9fc59403-dc6d-4011-830d-cf7d7c76a6d0.png)

AryNews would be https://6zklx4wryw9b-hls-live.5centscdn.com/arynewsweb/f7b44cfafd5c52223d5498196c8a2e7b.sdp/playlist.m3u8 / referer locked to https://live.arynews.tv/

P.S = we already have those channels, but they're flagged as offline.

As for the cricplay.xyz links, they work if you use Referer=https://cricplay2.xyz

**Reply by SSyar** [reply]
yess, but how to use m3u8 link in the playlist..

**Comment by BellezaEmporium** [comment]
> yess, but how to use m3u8 link in the playlist..

The method used would only work on VLC.

**Reply by SSyar** [reply]
> 
> The method used would only work on VLC.

elaborate.

**Reply by BellezaEmporium** [reply]
The "#EXTVLCOPT" line inside the playlist for these channels is VLC-specific, meaning only this player can read this as an optional header when doing the request. I also know that Kodi can read these lines and interpret it as additional parameters.

**Reply by SSyar** [reply]
can you be more specific ., (sorry for being n00b here) how would i add https://6zklx4wryw9b-hls-live.5centscdn.com/arynewsweb/f7b44cfafd5c52223d5498196c8a2e7b.sdp/playlist.m3u8 in the playlist.,

**Reply by BellezaEmporium** [reply]
```
#EXTINF:-1, AryNews
#EXTVLCOPT:http-referrer=https://live.arynews.tv/
https://6zklx4wryw9b-hls-live.5centscdn.com/arynewsweb/f7b44cfafd5c52223d5498196c8a2e7b.sdp/playlist.m3u8
```

**Comment by Sphinxroot** [comment]
https://anguillatv.net/watch-live/  will need  AI.m3u to be created

**Reply by BellezaEmporium** [reply]
Channel link would be : https://ott.streann.com/loadbalancer/services/public/channels/56bb43bfe4b01db401d1b065/playlist.m3u8 (or https://ott.streann.com/loadbalancer/services/public/channels/57c847612cdcf2eed4994471/playlist.m3u8 idk)

There might be a referer lock, but it's the first time I get a 403 on a Streann link.

**Comment by Sphinxroot** [comment]
https://tegostream.com/tv?channel=255 11 channels

Trinidad and Tobago Television https://www.ttt.live/stream/
https://tegostream.com/player/tego-player.php?broadcaster=ttt&channel=255
ABS TV http://abstvradio.com/live-streaming/
https://tegostream.com/player/tego-player.php?broadcaster=abstvradio&channel=854

**Reply by BellezaEmporium** [reply]
POST on https://tegostream.com/player/wms_auth.php

then add it on the end of the link.

**Comment by SSyar** [comment]
https://www.liveonlinetv.org/tvchannels.php

**Reply by BellezaEmporium** [reply]
GET on the channel you need, link should be packed in-script.
